include/flatbuffers/reflection.h

```diff
@@ -46,7 +46,9 @@ inline bool IsLong(reflection::BaseType t) {
 // Size of a basic type, don't use with structs.
 inline size_t GetTypeSize(reflection::BaseType base_type) {
   // This needs to correspond to the BaseType enum.
-  static size_t sizes[] = { 0, 1, 1, 1, 1, 2, 2, 4, 4, 8, 8, 4, 8, 4, 4, 4, 4 };
+  static size_t sizes[] = { 0, 1, 1, 1, 1, 2, 2, 4, 4, 8, 8, 4, 8, 4, 4, 4, 4, 0 };
```

Comment:
Add either inline comment saying new values must be left of the 0.

Reply:
New values don/t need to be added before this 0, but after it. This new 0 corresponds to Array, not to MaxBaseType. This is the motivation behind this change: I noticed that the addition of Array did not have a corresponding addition in this sizes array, so I opened a bug for it and I also suggested adding checks to prevent this discrepancy from re-occurring.

This is also why I named the new array entry CountBaseTypes rather than MaxBaseType - because it's not an actual type value and you're not expected to query its size, etc.

We could add a value for MaxBaseType, but that would just make things more confusing IMO, requiring commentary here. I don't see anyone getting confused and taking MaxBaseType for an actual type. If you prefer, I could add more comments in the fbs file above this entry.

Let me know which approach you prefer and I'll implement it.

Reply:
I see, I thought that the 0 was for the sentinel value (MaxBaseType), and hence my earlier comment on not using magic numbers. I still think a comment explaining why the value is zero is in needed since it is not clear looking at this code. I had to look at #5933 to see why this particular value was chosen.

I think something the following would really clarify the issue for future users.

```
static size_t sizes[] = {
  0, // None
  1, // UType
  1, // Bool
  ..
  0, // Array. 0 was chosen to prevent out-of-bounds errors.
}
```

And what happens if you do call `GetTypeSize(BaseType::MaxBaseType)` ? It looks like it would be an out-of-bounds error. So either we have a pre-condition check on that case, or just add another 0 to the end of the array to account for this.

Reply:
Sounds good.